### PR TITLE
More implementation of ReplayFileSystem

### PR DESCRIPTION
**Pull request by tvolkert**
Part of #11

### CODE REVIEW

**Comment by tvolkert** [comment]
@cbracken @goderbauer

File: lib/src/backends/record_replay/codecs.dart [hunk 1]
```diff
@@ -38,17 +44,50 @@ const Converter<dynamic, dynamic> kEntityTypeReviver =
 const Converter<dynamic, dynamic> kPathContextReviver =
     _PathContextCodec.kDecoder;
 
+/// Converter that will deserialize a [Uri].
+const Converter<dynamic, dynamic> kUriReviver = _UriCodec.kDecoder;
+
+/// Converter that will deserialize a [Encoding].
```
Comment (cbracken) [inline lib/src/backends/record_replay/codecs.dart]: nit: an

File: lib/src/backends/record_replay/codecs.dart [hunk 2]
```diff
@@ -332,46 +407,143 @@ class _EntityTypeCodec extends Codec<FileSystemEntityType, String> {
...
+
+class _FileSystemEntityReviver extends Converter<String, FileSystemEntity> {
+  final ReplayFileSystemImpl fileSystem;
+  const _FileSystemEntityReviver(this.fileSystem);
+
+  @override
+  FileSystemEntity convert(String input) {
+    if (input.contains('Directory')) {
+      return new ReplayDirectory(fileSystem, input);
+    } else if (input.contains('File')) {
```
Comment (cbracken) [inline lib/src/backends/record_replay/codecs.dart]: Here and above: would `endsWith()` work?

**Comment by goderbauer** [comment]
LGTM

File: lib/src/backends/record_replay/replay_io_sink.dart [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import 'codecs.dart';
+import 'replay_file_system.dart';
+import 'replay_proxy_mixin.dart';
+
+/// [IOSink] implementation that replays all invocation activity from a prior
+/// recording.
+class ReplayIOSink extends Object with ReplayProxyMixin implements IOSink {
+  final ReplayFileSystemImpl _fileSystem;
+
+  /// Creates a new `ReplayIOSink`.
```
Comment (cbracken) [inline lib/src/backends/record_replay/replay_io_sink.dart]: nit: `[ReplayIOSink]`

File: lib/src/backends/record_replay/replay_random_access_file.dart [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import 'replay_proxy_mixin.dart';
+
+/// [RandomAccessFile] implementation that replays all invocation activity from a prior
+/// recording.
+class ReplayRandomAccessFile extends Object
+    with ReplayProxyMixin
+    implements RandomAccessFile {
+  final ReplayFileSystemImpl _fileSystem;
+
+  /// Creates a new `ReplayIOSink`.
```
Comment (cbracken) [inline lib/src/backends/record_replay/replay_random_access_file.dart]: And here.

File: lib/src/backends/record_replay/codecs.dart (hunk 2 above)
Comment (tvolkert) [inline lib/src/backends/record_replay/codecs.dart]: Nope, the values will look like `RecordingDirectory@[0-9]+`, etc.

File: lib/src/backends/record_replay/codecs.dart (hunk 1 above)
Comment (tvolkert) [inline lib/src/backends/record_replay/codecs.dart]: Done in next PR

File: lib/src/backends/record_replay/replay_io_sink.dart (hunk 3 above)
Comment (tvolkert) [inline lib/src/backends/record_replay/replay_io_sink.dart]: Done in next PR

File: lib/src/backends/record_replay/replay_random_access_file.dart (hunk 4 above)
Comment (tvolkert) [inline lib/src/backends/record_replay/replay_random_access_file.dart]: Ditto

**Pull request merged**